cibuildwheel/linux.py

```diff
@@ -174,74 +177,79 @@ def build_on_docker(
             )
             sys.exit(1)
 
-        if build_options.before_build:
-            log.step("Running before_build...")
-            before_build_prepared = prepare_command(
-                build_options.before_build,
-                project=container_project_path,
-                package=container_package_dir,
-            )
-            docker.call(["sh", "-c", before_build_prepared], env=env)
-
-        log.step("Building wheel...")
-
-        temp_dir = PurePath("/tmp/cibuildwheel")
-        built_wheel_dir = temp_dir / "built_wheel"
-        docker.call(["rm", "-rf", built_wheel_dir])
-        docker.call(["mkdir", "-p", built_wheel_dir])
-
-        verbosity_flags = get_build_verbosity_extra_flags(build_options.build_verbosity)
-
-        if build_options.build_frontend == "pip":
-            docker.call(
-                [
-                    "python",
-                    "-m",
-                    "pip",
-                    "wheel",
-                    container_package_dir,
-                    f"--wheel-dir={built_wheel_dir}",
-                    "--no-deps",
-                    *verbosity_flags,
-                ],
-                env=env,
-            )
-        elif build_options.build_frontend == "build":
-            config_setting = " ".join(verbosity_flags)
-            docker.call(
-                [
-                    "python",
-                    "-m",
-                    "build",
-                    container_package_dir,
-                    "--wheel",
-                    f"--outdir={built_wheel_dir}",
-                    f"--config-setting={config_setting}",
-                ],
-                env=env,
-            )
+        abi3_wheel = find_abi3_wheel(built_wheels, config.identifier)
+        if abi3_wheel:
+            repaired_wheels = [abi3_wheel]
```

Comment:
It would be nice to add a log statement here, "Using previously built wheel filename.whl". You might have to close the previous log step first, otherwise it'll end up in the _Installing build tools_ step.

(same for the other platforms)

Reply:
done